autopush/router/simple.py

```diff
@@ -188,21 +194,42 @@ def _send_notification(self, uaid, node_id, notification):
                               "version": notification.version,
                               "data": notification.data})
         url = node_id + "/push/" + uaid
-        d = self.ap_settings.agent.request(
+        request = self.ap_settings.agent.request(
             "PUT",
             url.encode("utf8"),
             bodyProducer=FileBodyProducer(StringIO(payload)),
         )
-        d.addCallback(IgnoreBody.ignore)
-        return d
+        # Only time out actual calls, not mocks.
+        if isinstance(request, defer.Deferred):
+            timeout = reactor.callLater(self.ap_settings.router_timeout,
+                                        request.cancel)
```

Comment:
Well, [connectionTimeout](https://twistedmatrix.com/documents/current/api/twisted.web.client.Agent.html) is just how long to wait until a peer accepts a connection. The timeout is handed down to the connection protocol, which is essentially doing [the same sort of timeout](https://github.com/twisted/twisted/blob/twisted-17.1.0/src/twisted/internet/protocol.py#L271) as what I've got for the whole thing. 

Dropping the connectTimeout means that we don't timeout at all, which I don't think we want. Connection nodes bounce and we probably want to have some detection of that, right?